doc/cookbook/api-aggregation.md

```diff
@@ -0,0 +1,232 @@
+# API Aggregation
+
+- [API Aggregation](#api-aggregation)
+  - [Background](#background)
+  - [Example](#example)
+    - [Scenario 1: Simple aggregation](#scenario-1-simple-aggregation)
+    - [Scenario 2: Merge response body](#scenario-2-merge-response-body)
+    - [Scenario 3: Handle Failures](#scenario-3-handle-failures)
+  - [References](#references)
+
+## Background
+
+* API aggregation is a pattern to aggregate multiple individual requests into
+  a single request. This pattern is useful when a client must make multiple
+  calls to different backend systems to operate.[1]
+* Easegress provides filters `RequestBuilder` & `ResponseBuilder` for this
+  powerful feature.
+
```

Comment:
Suggest to add reference to filters.md request builder and response so that users won't get lost by these examples.